File: light-bulb.swift
import Foundation

let totalExpenses: Double = 61.0

let lampsRoom1 = 10
let daysRoom1 = 21
let hoursRoom1 = 12

let lampsRoom2 = 20
let daysRoom2 = 30
let hoursRoom2 = 8

let totalLamps = lampsRoom1 + lampsRoom2
let totalDays = daysRoom1 + daysRoom2
let totalHours = hoursRoom1 + hoursRoom2

let expensesRoom1 = (Double(lampsRoom1) / Double(totalLamps)) * 
                    (Double(daysRoom1) / Double(totalDays)) * 
                    (Double(hoursRoom1) / Double(totalHours)) * 
                    totalExpenses

let expensesRoom2 = (Double(lampsRoom2) / Double(totalLamps)) * 
                    (Double(daysRoom2) / Double(totalDays)) * 
                    (Double(hoursRoom2) / Double(totalHours)) * 
                    totalExpenses

print("Расходы на освещение первого помещения: \(expensesRoom1) рублей")
print("Расходы на освещение второго помещения: \(expensesRoom2) рублей")

File: light-bulb.rs
fn main() {
    let total_expenses: f64 = 61.0;
    
    let lamps_room1 = 10;
    let days_room1 = 21;
    let hours_room1 = 12;
    
    let lamps_room2 = 20;
    let days_room2 = 30;
    let hours_room2 = 8;
    
    let total_lamps = lamps_room1 + lamps_room2;
    let total_days = days_room1 + days_room2;
    let total_hours = hours_room1 + hours_room2;
    
    let expenses_room1 = (lamps_room1 as f64 / total_lamps as f64) * 
                         (days_room1 as f64 / total_days as f64) * 
                         (hours_room1 as f64 / total_hours as f64) * 
                         total_expenses;
    
    let expenses_room2 = (lamps_room2 as f64 / total_lamps as f64) * 
                         (days_room2 as f64 / total_days as f64) * 
                         (hours_room2 as f64 / total_hours as f64) * 
                         total_expenses;
    
    println!("Расходы на освещение первого помещения: {} рублей", expenses_room1);
    println!("Расходы на освещение второго помещения: {} рублей", expenses_room2);
}
